### DISCUSSION TITLE
Having the buildbot master listen on both IPv4 and IPv6

### DISCUSSION THREAD

**Discussion by levitte** [buildbot/buildbot#6279] [Q&A] [answered]
Are there ways to do this?  It seems difficult to do with the usual "tcp:" or "ssl:" connection strings, but what of the others that twisted supports.  Does anyone have any experience with that?

It would of course be nice if it was possible to have the build master listen on multiple ports, like this:

``` python
c = {
    'protocols': {
        'pb': {
            'port': [
                'ssl:port=9989:interface=0.0.0.0:certKey=certKey.pem:privateKey=privateKey.pem:extraCertChain=chain.pem',
                'ssl:port=9989:interface=\:\::certKey=certKey.pem:privateKey=privateKey.pem:extraCertChain=chain.pem',
           ]
        }
    }
}
```

(it seems that with current twisted, the "ssl:" endpoint type can really be used for both IPv4 and IPv6)

**Comment by p12tic** [comment, marked as answer]
I think this is a good feature request. With ipv6 becoming more and more important we'll definitely need this feature.

**Comment by fche** [comment]
`c['protocols'] = {'pb': {'port': r'tcp:9989:interface=\:\:'}}
`
